examples/store-and-load-results.ipynb

```diff
@@ -0,0 +1,387 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from skopt import gp_minimize\n",
+    "\n",
+    "noise_level = 0.1\n",
+    "\n",
+    "def obj_fun(x, noise_level=noise_level):\n",
+    "    return np.sin(5 * x[0]) * (1 - np.tanh(x[0] ** 2)) + np.random.randn() * noise_level\n",
+    "\n",
+    "res = gp_minimize(obj_fun,            # the function to minimize\n",
+    "                  [(-2.0, 2.0)],      # the bounds on each dimension of x\n",
+    "                  x0=[0.],            # the starting point\n",
+    "                  acq_func=\"LCB\",     # the acquisition function (optional)\n",
+    "                  n_calls=15,         # the number of evaluations of f including at x0\n",
+    "                  n_random_starts=0,  # the number of random initialization points\n",
+    "                  random_state=777)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As long as your Python session is active, you can access all the optimization results via the `res` object.\n",
+    "\n",
+    "So how can you store this data in a file? `skopt` conveniently provides functions `skopt.dump()` and `skopt.load()` that handle this for you. We will show how to use them for storing and loading results. Then we will also present two alternatives if you want more control over serialization. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using `skopt.dump()` and `skopt.load()`\n",
+    "\n",
+    "For storing optimization results into a file, call the `skopt.dump()` function:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from skopt import dump, load\n",
+    "\n",
+    "dump(res, 'result.pkl')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And load from file using `skopt.load()`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "-0.17487957729512074"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "res_loaded = load('result.pkl')\n",
+    "\n",
+    "res_loaded.fun"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`skopt.dump()` and `skopt.load()` use the `pickle` module internally. Notice that if the objective function is not serializable, `skopt.dump()` will raise a warning and store a copy of the optimization results object without the objective function. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Custom alternatives\n",
```

Comment:
Is this section necessary? Not sure we should be explaining about pickle or joblib here (especially since we are not doing anything more other than a thin wrapper over these)